src/Components/CreateImageWizardV2/common/TypeAheadSelect.tsx

```diff
@@ -0,0 +1,390 @@
+  inputOptions: [string | number, string][] | string[]
+): SelectOptionProps[] {
+  if (inputOptions.length === 0) {
+    return [
+      {
+        isDisabled: true,
+        children: `No results"`,
+        value: 'no results',
+      },
+    ];
+  }
+  return inputOptions.map((input) => {
+    if (typeof input === 'string') {
+      return { isDisabled: false, value: input, children: input };
+    }
+    return { isDisabled: false, value: input[0], children: input[1] };
+  });
+}
+
+function getLabel(
+  val: string | number,
+  inputOptions: [string | number, string][] | string[]
+): string {
+  if (typeof val === 'string') {
+    return val;
+  }
+  let ret: string = '';
+  inputOptions.forEach((input) => {
+    if (input[0] === val) {
+      ret = input[1];
+    }
+  });
+  return ret;
+}
+
+type TypeAheadSelectPropType<T extends string | number> = {
+  inputOptions: [T, string][] | string[];
+  fieldID: string;
+  placeholderText: string;
+  selected: [T, string] | string;
+  setSelected: Dispatch<SetStateAction<[T, string] | string>>;
+  isFetching?: boolean;
+  isError?: boolean;
+  refetch?: () => QueryActionCreatorResult<
+    QueryDefinition<
+      GetSourceListApiArg,
+      BaseQueryFn<
+        string | FetchArgs,
+        unknown,
+        FetchBaseQueryError,
+        {},
+        FetchBaseQueryMeta
+      >,
+      never,
+      V1ListSourceResponse,
+      'provisioningApi'
+    >
+  >;
+};
+
+/**
+ * Enhanced TypeAheadSelect from the pattern fly
+ * pattern https://www.patternfly.org/components/menus/select#typeahead
+ *
+ * The TypeAheadSelect component allows a user to select an element form a list
+ * of options. The user can also filter the elements by entering some text in
+ * the text input.
+ *
+ * The component can function in two different modes. Either in a mode where the
+ * key and value of the selectable items are the same, or in a mode where they
+ * and value are different. In the second operational mode, the key can be some
+ * number and the value a label of any kind.
+ *
+ * @param inputOptions a list of elements to be turned into select options.
+ * Either be a list of tuples or a list of string depending on the mode the
+ * component must operate in. If a tuple is passed, the first element of the
+ * tuple will end up being the key and the second element will be the value.
+ *
+ * @param selected the currently selected element. Either be a tuple or a string
+ * depending of the operating mode.
+ * @param setSelected a function to update the selected value.
+ *
+ * @param isFetching if provided and set to true, the component will show a
+ * loading spinner when the data is on the fly.
+ * @param isError if provided and set to true, the component will disable
+ * itself.
+ * @param refetch if provided, the component will use the refetch function to
+ * ask for a refresh of the data that makes the select options.
+ */
+function TypeAheadSelect<T extends string | number>({
+  inputOptions,
```

Comment:
I think it would be a bit more "React-y" if the `TypeAheadSelect` took a `children` prop instead of `inputOptions`, where the children were the `<SelectOption>`s.